### PR TITLE
Version 2

### PR DESCRIPTION
**Pull request by Permutatrix**
Yes, I finally made it. Whoop-diddle-dee-doo. Have fun, @operandom.

### CODE REVIEW

**Comment by operandom** [comment]
Great! I'll test it ! :)

File: index.js [hunk 1]
```diff
@@ -1,68 +1,122 @@
...
+        code: file.contents.toString(),
+        map:  file.sourceMap
+      };
+    } else {
+      wonderland[file.path] = file.contents.toString();
+    }
 
-    if (file.isStream()) {
-      return callback(new PluginError(PLUGIN_NAME, 'Streaming not supported'));
+    if (!entry && options.entry && hypothetical({ files: wonderland, allowRealFiles: true }).resolveId(options.entry)) {
```
Comment (operandom) [inline index.js]: Could be great to set gulp.src as entry if no entry is given and allow glob for entry.

Comment (Permutatrix) [inline index.js]: gulp-rollup is a changed package, @operandom. It swore never to go back to its old ways. So no using gulp.src as a globber. I will make it possible to have multiple entry files, though you'll have to use another package to do the globbing.

**Comment by operandom** [comment]
@Permutatrix I didn't mean a glob to math files on the filesystem but to match the gulp.src files.

**Comment by Permutatrix** [comment]
@operandom That's possible with a little effort.

```js
var gulp       = require('gulp'),
    rollup     = require('gulp-rollup'),
    Minimatch  = require('minimatch').Minimatch,
    through2   = require('through2');

var pattern = "entry-*.js";

gulp.task('bundle', function() {
  var modules = gulp.src('./src/**/*.js')
    //.pipe(...);
  var entry = new Promise(function(resolve, reject) {
    var files = [], match = new Minimatch(pattern, { matchBase: true });
    modules = modules.pipe(through2.obj(
      function transform(file, enc, cb) {
        if(match.match(file.path)) {
          files.push(file.path);
        }
        cb(null, file);
      },
      function flush(cb) {
        resolve(files);
        cb();
      }
    ));
  });
  modules
    .pipe(rollup({
      entry: entry
    }))
    .pipe(gulp.dest('./dist'));
});
```

That code could probably be improved, but it works. (Streams are the worst.) Maybe I'll make it possible to pass a callback function to `options.entry` that takes the files as input, but not today. That's super niche anyway as far as I can tell... This whole package is niche, or it ought to be, at any rate.

**Pull request merged**